### ISSUE TITLE
k-space preconditioning for pdhg

### ISSUE THREAD

**Issue by fzimmermann89** [PTB-MR/mrpro#545]
`Accelerating Non-Cartesian MRI Reconstruction Convergence using k-space Preconditioning` by `Frank Ong, Martin Uecker, and Michael Lustig` [arxiv](https://arxiv.org/pdf/1902.09657) suggests to use pdhg with k-space preconditoning.

For a diagonal precondition as proposed, this would in practice resulting the dual stepsize sigma being a tensor in our implementation.
See also how to do their experiments https://github.com/mikgroup/kspace_precond/blob/master/Sense%20Reconstruction.ipynb -- they set sigpy's pdhg stepsize to the precondition 

We would also need their methodof obtaining the single- or multi-channel preconditioner.

**Comment by koflera** [comment]
> `Accelerating Non-Cartesian MRI Reconstruction Convergence using k-space Preconditioning` by `Frank Ong, Martin Uecker, and Michael Lustig` [arxiv](https://arxiv.org/pdf/1902.09657) suggests to use pdhg with k-space preconditoning.
> 
> For a diagonal precondition as proposed, this would in practice resulting the dual stepsize sigma being a tensor in our implementation. See also how to do their experiments https://github.com/mikgroup/kspace_precond/blob/master/Sense%20Reconstruction.ipynb -- they set sigpy's pdhg stepsize to the precondition
> 
> We would also need their methodof obtaining the single- or multi-channel preconditioner.

I agree that would be of high interest... for example, for our TV-unrolling, it would be nice to be able to only use a few iterations of PDHG and already have converged to a decent solution. This would make the approach applicable to non-Cartesian MRI as well, which we on purpose did not consider yet because of those reasons.
